pimd/pim_bsm.c

```diff
@@ -285,15 +285,15 @@ static int pim_on_g2rp_timer(struct thread *t)
 	struct rp_info *rp_info;
 	struct route_node *rn;
 	uint16_t elapse;
-	struct in_addr bsrp_addr;
+	pim_addr bsrp_addr;
 
 	bsrp = THREAD_ARG(t);
 	THREAD_OFF(bsrp->g2rp_timer);
 	bsgrp_node = bsrp->bsgrp_node;
 
 	/* elapse time is the hold time of expired node */
 	elapse = bsrp->rp_holdtime;
-	bsrp_addr = bsrp->rp_address;
+	memcpy(&bsrp_addr, &(bsrp->rp_address), sizeof(pim_addr));
```

Comment:
No need for memcpy, `bsrp_addr = bsrp->rp_address;` works perfectly fine

(+ repeated below)

Reply:
Done